src/helpers/glucose-and-meals/sleep.ts

```diff
@@ -0,0 +1,34 @@
+import MyDataHelps from '@careevolution/mydatahelps-js';
+import { getMaxValue } from './util';
+import { totalSleepMinutes } from '../daily-data-providers/fitbit-sleep-v2';
+import getDayKey from '../get-day-key';
+import { asleepTime } from '../daily-data-providers/apple-health-sleep';
+import { endOfDay, startOfDay } from 'date-fns';
+
+export function fitbitSleepMinutesProvider(date: Date): Promise<number | undefined> {
+    return totalSleepMinutes(startOfDay(date), endOfDay(date)).then(sleepData => {
+        let dayKey = getDayKey(date);
+        return sleepData.hasOwnProperty(dayKey) ? sleepData[dayKey] : undefined;
+    }, () => undefined);
+}
+
+export function appleHealthSleepMinutesProvider(date: Date): Promise<number | undefined> {
+    return asleepTime(startOfDay(date), endOfDay(date)).then(sleepData => {
+        let dayKey = getDayKey(date);
+        return sleepData.hasOwnProperty(dayKey) ? sleepData[dayKey] : undefined;
+    }, () => undefined);
+}
+
+export async function getSleepMinutes(date: Date): Promise<number | undefined> {
+    let providers: Promise<number | undefined>[] = [];
+
+    let settings = await MyDataHelps.getDataCollectionSettings();
+    if (settings.fitbitEnabled) {
+        providers.push(fitbitSleepMinutesProvider(date));
+    }
+    if (settings.queryableDeviceDataTypes.find(s => s.namespace == 'AppleHealth' && s.type == 'SleepAnalysisInterval')) {
+        providers.push(appleHealthSleepMinutesProvider(date));
+    }
+
+    return providers.length > 0 ? await getMaxValue(providers) : undefined;
```

Comment:
It looks like this is the only logic thing that is why you're making your own provider here instead of using the provider in `combined-sleep.ts`?  Is there a reason that you made a new one instead of adjusting that provider so that it can have some sort of adjustable "accumulation" strategy?  

I ask partially because I notice that the new provider you're adding here doesn't touch Garmin data, and that might be unexpected.

Reply:
The main reason was that this uses a hybrid of V2 (for Fitbit) and V1 (for Apple Health).

The existing `combined-sleep.ts` uses V1 for both.  I haven't had a chance to examine how the Apple Health sleep data looks in V2, but I could probably take a look at that to create a V2 Apple Health Sleep query and then a new generic `combined-sleep-v2.ts`.

Good catch on Garmin.  Yes, I need to add that back, too.  This query evolved a little bit, since it was originally going to return half hour bucketed sleep values (like steps), and I was going to punt on Garmin for now, but then we reverted back to just a single daily value for the time being.  I'll get Garmin (using V1) added back.

Once a generic `combined-sleep-v2.ts` exists, this file can probably go away.  The `getMaxValue` accumulation strategy is already in use by the V1 combined sleep query, I just refactored it into a more explicitly named helper function.

Reply:
I've added support for V1 Garmin sleep minutes.

I'll take a look at what it would take to create V2 versions of the Apple Health and Garmin sleep queries (and a `combined-sleep-v2.ts`).

Reply:
I went ahead and created the V2 Apple Health and Garmin sleep queries, as well as a `combined-sleep-v2.ts`, which this file now references.